cinema/views.py

```diff
@@ -43,10 +48,37 @@ def get_serializer_class(self):
 
         return MovieSerializer
 
+    def filter_queryset(self, queryset):
+        """
+        Apply filters for movies based on title, genres, and actors.
+        """
+        # Get query parameters
+        title = self.request.query_params.get("title", None)
+        genres = self.request.query_params.get("genres", None)
+        actors = self.request.query_params.get("actors", None)
+
+        # Filter by title (case-insensitive, partial match)
+        if title:
+            queryset = queryset.filter(title__icontains=title)
+
+        # Filter by genres (comma-separated IDs)
+        if genres:
+            genre_ids = [int(id) for id in genres.split(",") if id.isdigit()]
+            queryset = queryset.filter(genres__id__in=genre_ids)
+
+        # Filter by actors (comma-separated IDs)
+        if actors:
+            actor_ids = [int(id) for id in actors.split(",") if id.isdigit()]
+            queryset = queryset.filter(actors__id__in=actor_ids)
+
+        # Remove duplicates caused by many-to-many relationships
+        return queryset.distinct()
```

Comment:
The `distinct()` method is used to remove duplicates caused by many-to-many relationships, ensuring that the queryset returns unique results.